lib/Infer/ExhaustiveSynthesis.cpp

```diff
@@ -119,9 +175,9 @@ std::vector<Inst *> matchWidth(Inst *I, unsigned NewW, InstContext &IC) {
   return { I };
 }
 
-void addGuess(Inst *RHS, int MaxCost, std::vector<Inst *> &Guesses,
-              int &TooExpensive) {
-  if (souper::cost(RHS) < MaxCost)
+void addGuess(Inst *RHS, float MaxCost, std::vector<Inst *> &Guesses,
+              float &TooExpensive, CostFunctionType &Cost) {
```

Comment:
rather than passing the cost function around a lot, investigate whether there's a better place to stash this pointer, for example in one of the context objects